pkg/document/json/json.go

```diff
@@ -37,3 +43,80 @@ func toOriginal(elem crdt.Element) crdt.Element {
 
 	panic("unsupported type")
 }
+
+func buildCRDTElement(
+	context *change.Context,
+	value interface{},
+	ticket *time.Ticket,
+) crdt.Element {
+	switch elem := value.(type) {
+	// primitive type
+	case string, int, int32, int64, float32, float64, []byte, bool, gotime.Time:
+		primitive, err := crdt.NewPrimitive(elem, ticket)
+		if err != nil {
+			panic(err)
+		}
+		return primitive
+
+	case *TreeNode:
+
+		return NewTree(context, crdt.NewTree(buildRoot(context, elem, ticket), ticket))
+
+	case *Text:
+		return NewText(context, crdt.NewText(crdt.NewRGATreeSplit(crdt.InitialTextNode()), ticket))
+	case *TempCounter:
+		switch elem.valueType {
+		case crdt.IntegerCnt:
+			counter, err := crdt.NewCounter(crdt.IntegerCnt, elem.value, ticket)
+			if err != nil {
+				panic(err)
+			}
+			return NewCounter(
+				context,
+				counter,
+			)
+		case crdt.LongCnt:
+			counter, err := crdt.NewCounter(crdt.LongCnt, elem.value, ticket)
+			if err != nil {
+				panic(err)
+			}
+			return NewCounter(
+				context,
+				counter,
+			)
+		default:
+			panic("unsupported type")
+		}
+
+	case []interface{}:
+		array := NewArray(context, crdt.NewArray(crdt.NewRGATreeList(), ticket))
+		for _, v := range elem {
+			ticket := context.IssueTimeTicket()
+			value := buildCRDTElement(context, v, ticket)
+			value = toOriginal(value)
+			if err := array.InsertAfter(array.LastCreatedAt(), value); err != nil {
+				panic(err)
+			}
+			array.context.RegisterElement(value)
```

Comment:
I consider the two choice
1. when create NewArray or NewObject, register their child elements. like js
2. register in buildCRDTElement like now

I thought first option is better than second, 
becuase it will be utilized in the case of SetNewArray.

But there was no idea to implement method overloading in go to not destroying the previous structure. so i choose the second option.

https://github.com/yorkie-team/yorkie/blob/3876b8f4f6ba171ea365ec77514b00fe99b71ec3/pkg/document/crdt/array.go#L33

If variable factors are used, it may be possible while maintaining the existing structure.
I'm not used to Golang yet, so I think I need to check it out.

```go

func NewArray(elements *RGATreeList, createdAt *time.Ticket, value ...crdt.element)
```